compose/foundation/foundation/src/commonMain/kotlin/androidx/compose/foundation/text/selection/TextFieldSelectionManager.kt

```diff
@@ -311,6 +311,9 @@ internal class TextFieldSelectionManager(
                     isStartHandle = false,
                     adjustment = adjustment
                 )
+
+                undoManager?.forceNextSnapshot()
```

Comment:
Let's add unit test to cover this case

Reply:
I struggled to create one for this, because it requires cursor interaction and hence some sort of layout feedback from compose UI. Can you help me with it?

Reply:
I can hardcode it though.